tests/aws-cpp-sdk-s3-crt-integration-tests/RunTestsWithMemTracer.cpp

```diff
@@ -0,0 +1,29 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <gtest/gtest.h>
+#include <aws/core/Aws.h>
+#include <aws/testing/platform/PlatformTesting.h>
+#include <aws/testing/TestingEnvironment.h>
+#include <aws/testing/MemoryTesting.h>
+
+int main(int argc, char** argv) {
+    Aws::Testing::SetDefaultSigPipeHandler();
+    Aws::SDKOptions options;
+    options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
+    CRTMemTracerMemorySystem memorySystem{};
+    options.memoryManagementOptions.memoryManager = &memorySystem;
+    Aws::Testing::InitPlatformTest(options);
+    Aws::Testing::ParseArgs(argc, argv);
+
+    Aws::InitAPI(options);
```

Comment:
I think Begin() goes here and End() right after "ShutdownAPI"

Reply:
this is copy and pasted more or less from what we [already do](https://github.com/aws/aws-sdk-cpp/blob/main/tests/testing-resources/include/aws/testing/MemoryTesting.h#L131) and how we use it in the [crt test already](https://github.com/aws/aws-sdk-cpp/blob/main/tests/aws-cpp-sdk-s3-crt-integration-tests/RunTests.cpp#L17)